src/main/java/edu/kit/hci/soli/controller/BookingsController.java

```diff
@@ -1,25 +1,119 @@
 package edu.kit.hci.soli.controller;
 
-
-import edu.kit.hci.soli.domain.LoginStateModel;
-import edu.kit.hci.soli.repository.VisitsRepository;
+import edu.kit.hci.soli.domain.*;
+import edu.kit.hci.soli.dto.KnownError;
+import edu.kit.hci.soli.dto.LoginStateModel;
+import edu.kit.hci.soli.service.BookingsService;
+import edu.kit.hci.soli.service.RoomService;
+import edu.kit.hci.soli.service.UserService;
 import jakarta.servlet.http.HttpServletResponse;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.security.core.annotation.AuthenticationPrincipal;
-import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import lombok.*;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import java.security.Principal;
+import org.springframework.web.bind.annotation.*;
 
+import java.security.Principal;
+import java.time.LocalDateTime;
+@Slf4j
 @Controller("/bookings")
 public class BookingsController {
 
-    @Autowired
-    private VisitsRepository visitsRepository;
+    private final BookingsService bookingsService;
+    private final RoomService roomService;
+    private final UserService userService;
+
+    public BookingsController(BookingsService bookingsService, RoomService roomService, UserService userService) {
+        this.bookingsService = bookingsService;
+        this.roomService = roomService;
+        this.userService = userService;
+    }
 
     @GetMapping("/bookings")
-    public String bookings(Model model, HttpServletResponse response, Principal principal) {
+    public String userBookings(Model model, HttpServletResponse response, Principal principal) {
+
+        User user = userService.getCurrentlyLoggedInUserByUserId(principal.getName());
```

Comment:
I find this much smoother... Especially since we dont toy around with the model. This is also afaik the spring boot way todo it...

Reply:
The advantage of digging this out through the model is that we avoid duplicating the computation we need to do for the shared layout, which really isn't significant.
As such, I agree that something like this is probably better since it is clearer.
That said: If we go this route, I think passing the principal to some method is preferable, since it improves type safety.